### PR TITLE
Added mean and niterations to sigma clipping

### PR DESCRIPTION
**Pull request by joshuawallace**
Added functionality to sigmaclip_magseries() so that clipping can be based around the mean instead of the median, and so iterative sigma clipping can be given a maximum number of iterations to perform.

### CODE REVIEW

File: astrobase/lcmath.py [hunk 1]
```diff
@@ -345,21 +400,21 @@ def sigclip_magseries(times, mags, errs,
         dimmingclip = sigclip[0]
         brighteningclip = sigclip[1]
 
-        if not iterative:
+        if not iterative and niterations == None:
```
Comment (waqasbhatti) [inline astrobase/lcmath.py]: Same as above

File: astrobase/lcmath.py [hunk 2]
```diff
@@ -373,43 +428,93 @@ def sigclip_magseries(times, mags, errs,
...
-                tsi = nottoodimind & nottoobrightind
-
-                # update the arrays
-                this_times = this_times[tsi]
-                this_mags = this_mags[tsi]
-                this_errs = this_errs[tsi]
-
-                # update delta and go to top of the loop
-                delta = this_size - this_mags.size
+            if niterations == None:
```
Comment (waqasbhatti) [inline astrobase/lcmath.py]: Here too

File: astrobase/lcmath.py [hunk 3]
```diff
@@ -291,16 +305,22 @@ def sigclip_magseries(times, mags, errs,
...
-    stddev_mag = (npmedian(npabs(fmags - median_mag))) * 1.483
+    # get the center value and stdev 
+    if meanormedian == 'median':  # stddev = 1.483 x MAD
+        center_mag = npmedian(fmags)
+        stddev_mag = (npmedian(npabs(fmags - center_mag))) * 1.483
+    elif meanormedian == 'mean':
+        center_mag = npmean(fmags)
+        stddev_mag = npstddev(fmags)
+    else:
+        LOGERROR("Unrecognized meanormedian value given to sigclip_magseries: " + str(meanormedian))
```
Comment (waqasbhatti) [inline astrobase/lcmath.py]: Better to use:
```python
LOGERROR("Unrecognized meanormedian value given to sigclip_magseries: %s " % meanormedian)
```
We should also deal with the weird `meanormedian` input here. I suggest changing the LOGERROR to LOGWARNING, setting `meanormedian` to 'median' and proceeding to process the LC after informing about the problem.

File: astrobase/lcmath.py [hunk 4]
```diff
@@ -291,16 +305,22 @@ def sigclip_magseries(times, mags, errs,
...
 
     # sigclip next for a single sigclip value
     if sigclip and isinstance(sigclip,float):
 
-        if not iterative:
+        if not iterative and niterations == None:
```
[the remark this answers is not in the document]
Comment (joshuawallace) [reply astrobase/lcmath.py]: How do these changes work?  Do I put them in or do you put them in?

[the remark this answers is not in the document]
Comment (waqasbhatti) [reply astrobase/lcmath.py]: You can put them in. I think you can commit and then push to this branch: https://github.com/joshuawallace/astrobase/tree/patch-2.

**Comment by joshuawallace** [comment]
I used the updated sigclip_magseries with a real light curve using both meanormedian='mean' and niterations=3 and got behavior as expected, verified with an outside sigclip calculation.  Just a further verification.

**Comment by joshuawallace** [comment]
I made the requested changes and committed to the patch-2 branch on my fork.  I don't see a pull request button popping up anywhere, does the new commit automatically appear in the pull request?

**Comment by waqasbhatti** [comment]
Looks good now. Thanks!

**Pull request merged**